src/Ocelot/LoadBalancer/LoadBalancers/RoundRobin.cs

```diff
@@ -27,6 +24,12 @@ public async Task<Response<ServiceHostAndPort>> Lease(HttpContext httpContext)
             var services = await _services();
             lock (_lock)
             {
+                if (services.Count < 1)
+                {
+                    // When the downstream service is not found, LeastConnection prompts Warn and RoundRobin throws an exception, eg: /favicon.ico
```

Comment:
Truly speaking this case of icon should not be processed by this method.
Can we filter such cases in upper contexts?